pkg/infoprovider/vhostNetInfoProvider.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package infoprovider
+
+import (
+	"os"
+
+	"github.com/golang/glog"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+/*
+   VhostNetInfoProvider wraps any DeviceInfoProvider and adds a vhost-net device
+*/
+type vhostNetInfoProvider struct {
+}
+
+// NewVhostNetInfoProvider returns a new Vhost Information Provider
+func NewVhostNetInfoProvider() types.DeviceInfoProvider {
+	return &vhostNetInfoProvider{}
+}
+
+// VhostNetDeviceExist returns true if /dev/vhost-net exists
+func VhostNetDeviceExist() bool {
+	_, err := os.Stat("/dev/vhost-net")
```

Comment:
Actually, this method could return false for different reasons. Please check if error is ErrNotExist or rename the method

Reply:
Frankly, this functions aren't really usable, except that they just hide call to `Stat`. Mb it's better to call it directly?